File: ryan/230131_study/프로_42883_큰수만들기.py
def solution(number, k):
    p = 0
    for _ in range(k):
        for idx in range(max(p-1, 0), len(number) - 1):  # range( len(number-1))
            if number[idx] < number[idx + 1]:
                number = number[:idx] + number[idx + 1:]
                p = idx
                break
        else:
            number = number[:-1]

    return number

File: James/230131_study/프로_42883_큰수만들기.java
import java.util.*;

class Solution {
    public String solution(String number, int k) {
        String answer = "";
        StringBuilder sb = new StringBuilder();
        int index = 0;
        int max;
        int numberLength = number.length() - k;
        
        for(int i = 0; i < numberLength; i++) {
            max = 0;
            for(int j = index; j < k + i + 1; j++) {
                if(max < number.charAt(j) - '0') {
                    max = number.charAt(j) - '0';
                    index = j + 1;
                }
            }
            sb.append(max);
            // System.out.println(sb.toString());
        }
        
        answer = sb.toString();
        
        return answer;
    }
}
